rfcs/text/semantic-release.md

```diff
@@ -0,0 +1,257 @@
+
+# How we teach this
+
+The proposed tooling is well established in the JavaScript community. For carbon developers
+ can use `npm run commit` to launch [`commitizen`](https://github.com/commitizen/cz-cli) which will guide them through the process.
+![](./commitizen.svg)
+
+| Type           | Description                                                                                                 | Release Type |
+|----------------|-------------------------------------------------------------------------------------------------------------|--------------|
+| `feat:`        | A new feature                                                                                               | minor        |
+| `fix:`         | A bug fix                                                                                                   | patch        |
+| `docs:`        | Documentation only changes                                                                                  | :x:          |
+| `style:`       | Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)      | :x:          |
+| `refactor:`    | A code change that neither fixes a bug nor adds a feature                                                   | :x:          |
+| `perf:`        | A code change that improves performance                                                                     | patch        |
+| `test:`        | Adding missing tests or correcting existing tests                                                           | :x:          |
+| `build:`       | Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)         | :x:          |
+| `ci:`          | Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs) | :x:          |
+| `chore:`       | Other changes that don't modify src or test files                                                           | :x:          |
+| `revert:`      | Reverts a previous commit                                                                                   | patch        |
+
+This behaviour is defined in [@semantic-release/commit-analyzer](https://github.com/semantic-release/commit-analyzer/blob/95a21dc7b09b7bff753d3d4a00e684b69a81df8c/lib/default-release-rules.js).
+
+Please note that `improvement:` is **not** supported, there is an [open pull request](https://github.com/commitizen/conventional-commit-types/pull/16) to remove it from [`commitizen`](https://github.com/commitizen/cz-cli).
+
+If we use [`git revert`](https://git-scm.com/docs/git-revert) we will use the [`--no-commit`](https://git-scm.com/docs/git-revert#Documentation/git-revert.txt---no-commit) flag if we're reverting a `BREAKING CHANGE` because this would also be a breaking change, or, if we're reverting a new feature. We will then use `commitizen` to write a commit that includes a reference to the `BREAKING CHANGE`.
```

Comment:
When using `git revert`, wouldn't we **always** want to use the `--no-commit` flag? (If we don't use that flag, then `git revert` will auto-generate a commit message, which won't follow the Conventional Commits spec.)

Would this be better?
```suggestion
If we use [`git revert`](https://git-scm.com/docs/git-revert), we will use the [`--no-commit`](https://git-scm.com/docs/git-revert#Documentation/git-revert.txt---no-commit) flag (to prevent auto-generation of a commit message) and then use `commitizen` to write the commit message.
```